pkg/controller/configmap/config.go

```diff
@@ -137,22 +137,12 @@ func (adaptor *ConfigMapK8s) validateConfigMap(configmap *corev1.ConfigMap) []er
 		validateConfig(cfg, "nsx_v3", "top_tier_router") != nil {
 		appendErrorIfNotNil(&errs, errors.Errorf("failed to get tier0_gateway or top_tier_router"))
 	}
-	// Check MTU value
-	mtu := cfg.Section("nsx_node_agent").Key("mtu").Value()
-	_, err = strconv.Atoi(mtu)
-	if err != nil {
-		appendErrorIfNotNil(&errs, errors.Wrapf(err, "mtu is invalid"))
-	}
-	// Check DEFAULT section log_file option
-	if cfg.Section("DEFAULT").Key("log_file").Value() != "" {
-		log.Info(fmt.Sprintf("DEFAULT section option: log_file is not allowed to change, current value:%s is reverted back to default value:None",
-			cfg.Section("DEFAULT").Key("log_file").Value()))
-		// Revert DEFAULT log_file to default value
-		fillDefault(cfg, "DEFAULT", "log_file", "", true)
-		// Write config back to ConfigMap data
-		(*data)[operatortypes.ConfigMapDataKey], err = iniWriteToString(cfg)
-		appendErrorIfNotNil(&errs, err)
-	}
+
+	errs = append(errs, validateNodeAgentOptions(configmap, cfg)...)
+	errs = append(errs, validateCommonOptions(configmap, cfg)...)
+	errs = append(errs, validateMPOptions(configmap, cfg)...)
+	errs = append(errs, validateWCPOptions(configmap, cfg)...)
+	errs = append(errs, validateTASOptions(configmap, cfg)...)
```

Comment:
Let's do not distinguish other platforms, we can put them all in one blocklist.

Reply:
You mean merge validateWCPOptions and validateWCPOptions into one function?

Reply:
Yes, call it like validateConfigOptions. I feel it's better do not expose other platforms in operator code.